src/python/library/tritonclient/utils/cuda_shared_memory/__init__.py

```diff
@@ -173,20 +181,27 @@ def set_shared_memory_region(cuda_shm_handle, input_values):
                 "input_values must be specified as a list/tuple of numpy arrays"
             )
 
+    if 'cpu' == device:
+        fptr = _ccudashm_shared_memory_region_set
+    elif 'gpu' == device:
+        fptr = _ccudashm_shared_memory_region_set_dptr
+    else:
+        _raise_error(
+            "unsupported device type: cpu and gpu are supported only"
+        )
+
     offset_current = 0
     for input_value in input_values:
```

Comment:
Is the input_values a list of cudaptr or a single cudaptr? I think a single cuda device ptr is suffcient.

Reply:
It's a single cudaptr.